docs/developer-guide/udf.rst

```diff
@@ -233,20 +234,30 @@ of the UDF does, for example:
 UdfParameter Annotation
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``@UdfParameter`` annotation is optional and is applied to the parameters of methods annotated with
-``@Udf``. KSQL will use the additional information in the ``@UdfParameter`` annotation to provide
-users with richer information about the method when, for example, they execute
-``DESCRIBE FUNCTION`` on the method.
-
-The annotation has two parameters: ``value`` is the name of the parameter and ``description`` which
-can be used to better describe what the parameter does, for example:
+The ``@UdfParameter`` annotation is applied to parameters of methods annotated with ``@Udf``. KSQL
+will use the additional information in the ``@UdfParameter`` annotation to specify the parameter
+schema (if it cannot be inferred from the Java type) or to provide users with richer information
+about the method when, for example, they execute ``DESCRIBE FUNCTION`` on the method.
+
++------------+------------------------------+------------------------+
+| Field      | Description                  | Required               |
++============+==============================+========================+
+| value      | The case-insensitive name of | No                     |
+|            | the parameter                |                        |
++------------+------------------------------+------------------------+
+| description| A string describing generally| No                     |
+|            | what the parameter represents|                        |
++------------+------------------------------+------------------------+
+| schema     | The KSQL schema for the      | For complex types      |
+|            | parameter.                   | such as STRUCT         |
++------------+------------------------------+------------------------+
 
 .. code:: java
 
     @Udf
-    public String substring(
-       @UdfParameter("str") final String str,
-       @UdfParameter(value = "pos", description = "Starting position of the substring") final int pos)
+    public boolean livesInRegion(
+       @UdfParameter(value = "zipcode", description = "a US postal code") final String zipcode,
+       @UdfParameter(schema = "STRUCT<ZIP STRING, NAME STRING>") final Struct employee)
```

Comment:
How strict are these input schemas? For example, with this UDF, would I be allowed to pass in a `STRUCT<ZIP STRING>` or `STRUCT<ZIP STRING, NAME STRING, AGE BIGINT>` for this parameter or no? What about `STRUCT<NAME STRING, ZIP STRING>`?

Reply:
They are unbounded - i.e. you can have whatever crazy nested struct that your heart desires.

Reply:
Sorry, don't think my question made it across. I was asking about what parameters this particular UDF (as defined) would accept during runtime. For example, with this UDF defined as is, would a user be able to call the UDF on fields `zip` and `person` where `person` has schema `STRUCT<NAME STRING, ZIP STRING>` (rather than `STRUCT<ZIP STRING, NAME STRING>`)?

Reply:
Ah! Now I got it. It is super strict - it will fail if you pass in anything other than the exact struct schema. See the tests for `UdfIndex`.